app/models/debt_filter.rb

```diff
@@ -0,0 +1,55 @@
+module DebtFilter
```

Comment:
This module has one public method, but 5 private methods that handle combining debts between the same people. I think this can be simplified a lot and would encourage you to spend some time refactoring it.